brightener.h

```diff
@@ -1,17 +1,31 @@
 #pragma once
 #include <cstdint>
+#include <memory>
+#include <algorithm>
+#include <random>
 
 struct Image {
-	int rows;
-	int columns;
-	uint8_t pixels[1024 * 1024]; // max 1k x 1k image
+    int rows;
+    int columns;
+    std::unique_ptr<uint8_t[]> pixels;
+
+    // Constructor to initialize rows, columns, and set all pixel values to random values between 0 and 255
+    Image(int r, int c) : rows(r), columns(c), pixels(std::make_unique<uint8_t[]>(r * c)) {
+        std::random_device rd;  // Seed for the random number engine
+        std::mt19937 gen(rd()); // Mersenne Twister random number engine
+        std::uniform_int_distribution<> dis(0, 255); // Uniform distribution between 0 and 255
+
+        for (int i = 0; i < r * c; ++i) {
+            pixels[i] = dis(gen); // Assign random value to each pixel
+        }
+    }
 };
 
 class ImageBrightener {
 private:
-	Image m_inputImage;
+    Image& m_inputImage;
```

Comment:
The `ImageBrightener` keeps a reference to `Image`. That imposes an "implicit" constraint, that the lifetime of the referred object needs to be at least as much as the ImageBrightener object. Why not use a `unique_ptr` or `shared_ptr` instead of a reference?